pkg/storage/stores/shipper/uploads/table.go

```diff
@@ -244,6 +259,16 @@ func (lt *Table) Write(ctx context.Context, writes local.TableWrites) error {
 // db files are named after the time shard i.e epoch of the truncated time.
 // If a db file does not exist for a shard it gets created.
 func (lt *Table) write(ctx context.Context, tm time.Time, writes local.TableWrites) error {
+	writeToBucket := defaultBucketName
+	if lt.makePerTenantBuckets {
```

Comment:
I wonder if we should only write to tenant bucket if the global bucket is empty ?

Reply:
We never write to files created before ingester restart so a file would either have a global bucket or per user bucket. I hope it also answers your question?
> what if we started building data in a table in the global bucket and switch ingester ? The code here does seems to loose the current data for reads.